sql/handler.cc

```diff
@@ -8126,6 +8131,9 @@ int handler::ha_update_row(const uchar *old_data, const uchar *new_data)
   DBUG_ASSERT(new_data == table->record[0]);
   DBUG_ASSERT(old_data == table->record[1]);
 
+  if (table->open_hlindexes_for_write())
+    return 1;
+
```

Comment:
strange, `ha_write_row()` doesn't do it, it opens the table once, not for every row. perhaps update/delete should do the same?